components/feature/search/src/test/java/mozilla/components/feature/search/SearchFeatureTest.kt

```diff
@@ -0,0 +1,114 @@
+package mozilla.components.feature.search
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.setMain
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.search.SearchRequest
+import mozilla.components.support.test.any
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.mock
+import org.junit.After
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+
+private const val SELECTED_TAB_ID = "1"
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+class SearchFeatureTest {
+
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    private lateinit var performSearch: (SearchRequest) -> Unit
+    private lateinit var store: BrowserStore
+    private lateinit var searchFeature: SearchFeature
+
+    @Before
+    fun before() {
+        Dispatchers.setMain(testDispatcher)
+        store = BrowserStore(
+            mockBrowserState(),
+            dispatcher = TestCoroutineDispatcher()
```

Comment:
You shouldn't need this here? (see comments above) :)

You're calling `joinBlocking` on the dispatching job anyway. Here's an example of another test using our `MainCoroutineRule`: https://github.com/mozilla-mobile/android-components/blob/master/components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionPopupFeatureTest.kt